app/src/main/java/com/firebase/uidemo/database/firestore/FirestorePagingActivity.java

```diff
@@ -84,34 +83,42 @@ protected void onBindViewHolder(@NonNull ItemViewHolder holder,
                                                     @NonNull Item model) {
                         holder.bind(model);
                     }
+                };
+        adapter.addLoadStateListener(new Function1<CombinedLoadStates, Unit>() {
```

Comment:
Seeing `Function1<>` 🤮  reminds me that we can probably convert the sample app to Kotlin once we go to 8.0!

Reply:
We definitely should!

Reply:
@samtstern actually, since we migrated to Java 8, this can actually be replaced with a lambda function:

```java
adapter.addLoadStateListener(states -> {
    // ...
}
```

So we don't really need `Function1<>` here. :sweat_smile: